src/OmniSharp.Abstractions/Services/IMemberSyntaxFeaturesDiscover.cs

```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace OmniSharp.Abstractions.Services
+{
+    public interface ISyntaxFeaturesDiscover
+    {
+        bool NeedSemanticModel { get; }
+        
+        IEnumerable<string> Discover(SyntaxNode node, SemanticModel semanticModel);
```

Comment:
Actually looking at this again, we might not be able to do that due to the to the SyntaxWalker...